torcheeg/trainers/classifier.py

```diff
@@ -266,4 +296,349 @@ def predict_step(self,
+
+        Args:
+            model (nn.Module): model (nn.Module): The classification model, which should provide the :obj:`".decoder()"` , :obj:`".predict_by_feature()"` method or the model  should contain decoder block(nn.Module), predict block(nn.Module) attributes and ensure this two attirbutes are named "decoder","predict_by_feature".
+            center_dim (int): The dimemsion of decoder output code whose mean values we can loosely regard as the "center". If you don't specify this value, trainer will automatically infer this value when we first pass dataloader into :obj:`trainer.fit()` or :obj:`trainer.test()` method. (defualt: :obj:`None`) 
+            num_classes (int, optional): The number of categories in the dataset. If :obj:`None`, the number of categories will be inferred from the attribute :obj:`num_classes` of the model. (defualt: :obj:`None`)
+            lammda (float): The weight of the center loss in total loss. (default: :obj:`5e-4`)
+            lr (float): The learning rate. (default: :obj:`0.001`)
+            weight_decay (float): The weight decay. (default: :obj:`0.0`)
+            devices (int): The number of devices to use. (default: :obj:`1`)
+            accelerator (str): The accelerator to use. Available options are: 'cpu', 'gpu'. (default: :obj:`"cpu"`)
+            metrics (list of str): The metrics to use. Available options are: 'precision', 'recall', 'f1score', 'accuracy'. (default: :obj:`['accuracy', 'precision', 'recall', 'f1score']`)
+
+        .. automethod:: fit
+        .. automethod:: test
+    '''
+    def __init__(self,
+                 model: nn.Module,
+                 center_dim: int = None,
+                 num_classes: int = None,
+                 lammda: float = 0.0005,
+                 lr: float = 1e-3,
+                 weight_decay: float = 0.0,
+                 devices: int = 1,
+                 accelerator: str = "cpu",
+                 metrics: List[str] = [
+                     'accuracy', 'precision', 'recall', 'f1score'
+                 ]):
+        
+        super(CLossClassifierTrainer,self).__init__(model, num_classes, lr, weight_decay, devices,
+                             accelerator, metrics,center_dim = center_dim) 
+        self.lammda = lammda
+        self.automatic_optimization = False
+        self.decoder,self.Fpredictor= (model.decoder, 
+                                       model.predict_by_feature)
+        self.center_dim = center_dim
+        self.__try_init_center_dim()
+    
+
+        
+
+        
+        
+    def __try_init_center_dim(self):
+        if not self.center_dim :
+            try:
+                self.center_dim = self.model.center_dim
+                if self.num_classes:
+                    self.Closs = CentersLoss(self.num_classes, self.center_dim)
+            except AttributeError:
+                pass
+        else:
+            if self.num_classes:
+                self.Closs = CentersLoss(self.num_classes, self.center_dim)
```

Comment:
Class names should be in CamelCase, while instances should be named using snake_case. E.g., `self.center_loss = CenterLoss(...)`